Allow custom npmconfig to be defined in settings.json

Summary: Node --userconfig and --globalconfig are not configurable, particularly leading to issues when using non default package registries.

When trying to install a language server or anything that utilizes Node, it uses a default blank config for the --userconfig and --globalconfig. This leads to it pulling its packages from `registry.npmjs.org` by default, without an option to specify a different location to pull node packages from. 

When behind proxies or at some work places, that registry may be blocked or you may want to pull packages from another source. 

# Proposal:

Since it looks like Zed bundles a node runtime to avoid issues and different node versions may return unexpected results, I don't think having a setting to specify the node path would be needed. But it would be great if we add an option in the user settings to specify the configs to use. 

Something like this being added to settings would work:
```json
{
      "node_runtime": {
          "custom_node_user_config": "path/to/config/here",
          "custom_node_global_config": "path/to/config/here"
  }
}
```

# Possible solution:

I did a bit of digging into the issue and it looks like a user specified path could be added here, where it will default to the blank file by default:

Zed calls node with the blankconfigs here:
https://github.com/zed-industries/zed/blob/eecbafb94e73afff185e3160200a60db191cc35c/crates/node_runtime/src/node_runtime.rs#L218-L224

As a temporary workaround I tried modifying the blank configs with a proper config but it looks like each time a command is run it will re-write the default blank config files:
https://github.com/zed-industries/zed/blob/eecbafb94e73afff185e3160200a60db191cc35c/crates/node_runtime/src/node_runtime.rs#L144-L172


I think this would be pretty simple to add and tried taking a look at it as well, but I have no prior rust experience. I did take a look at maybe adding something like this to the node_runtime.js file.
```rust
use std::str;

When trying to install a language server or anything that utilizes Node, it uses a default blank config for the --userconfig and --globalconfig. This leads to it pulling its packages from `registry.npmjs.org` by default, without an option to specify a different location to pull node packages from. 

When behind proxies or at some work places, that registry may be blocked or you may want to pull packages from another source. 

use serde::Deserialize;
use settings::Settings;

#[derive(Deserialize, Clone)]
pub struct NodeSettings {
    pub custom_node_user_config: Option<String>,
    pub custom_node_user_config: Option<String>,
    pub custom_node_global_config: Option<String>,
}

impl Settings for NodeSettings{
    const KEY: Option<&'static str> = Some("message_editor");
    type FileContent = NodeSettings;
    fn load(
        default_value: &Self::FileContent,
        user_values: &[&Self::FileContent],

        _: &mut gpui::AppContext,
    ) -> anyhow::Result<Self> {
        Self::load_via_json_merge(default_value, user_values)
    }
}
```

```
stderr: "npm ERR! code UNABLE_TO_GET_ISSUER_CERT_LOCALLY\nnpm ERR! errno UNABLE_TO_GET_ISSUER_CERT_LOCALLY\nnpm ERR! request to https://registry.npmjs.org/prettier failed, reason: unable to get local issuer certificate\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /Users/viu984/Library/Application Support/Zed/node/node-v18.15.0-darwin-arm64/cache/_logs/2024-07-25T14_39_08_962Z-debug-0.log\n"
```

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -499,12 +499,6 @@ impl SystemNodeRuntime {
         let scratch_dir = paths::support_dir().join("node");
         fs::create_dir(&scratch_dir).await.ok();
         fs::create_dir(scratch_dir.join("cache")).await.ok();
-        fs::write(scratch_dir.join("blank_user_npmrc"), [])
-            .await
-            .ok();
-        fs::write(scratch_dir.join("blank_global_npmrc"), [])
-            .await
-            .ok();
 
         let mut this = Self {
             node,

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -551,14 +551,6 @@ impl NodeRuntimeTrait for SystemNodeRuntime {
             .env(NODE_CA_CERTS_ENV_VAR, node_ca_certs)
             .arg(subcommand)
             .args(["--cache".into(), self.scratch_dir.join("cache")])
-            .args([
-                "--userconfig".into(),
-                self.scratch_dir.join("blank_user_npmrc"),
-            ])
-            .args([
-                "--globalconfig".into(),
-                self.scratch_dir.join("blank_global_npmrc"),
-            ])
             .args(args);
         configure_npm_command(&mut command, directory, proxy);
         let output = command.output().await?;

